consensus/src/dag/bootstrap.rs

```diff
@@ -415,15 +428,56 @@ impl DagBootstrapper {
             metadata_adapter,
             heuristic,
             100,
-            ChainHealthBackoffConfig::new(self.config.chain_backoff_config.clone()),
         ))
     }
 
+    fn build_anchor_election(
+        &self,
+    ) -> (
+        Arc<dyn AnchorElection>,
+        Option<Arc<LeaderReputationAdapter>>,
+        Option<Vec<CommitEvent>>,
```

Comment:
This is possibly dirty. I couldn't cleanly separate them in this PR.

Reply:
we probably should always populate the commit events (for the sake of completeness)

Reply:
to make it look better, we can create another trait to get voting power ratio and having a no-op one always returns 1 so here can return Arc\<dyn AnchorElection\>, Arc\<dyn CommitHistory\>, Vec\<CommitEvent\>

Reply:
I am not sure that we can populate always. How many events do we read? For `AnchorElectionMode::LeaderReputation` we have a config with num events to read, but not for RoundRobin Leader. Also, it seems unnecessary to read in the latter case?

Reply:
Also, introduced the CommitHistory trait.